### PR TITLE
Fix EbpfProgramType for sample extension.

### LINKED ISSUE #568: EbpfProgramType::platform_specific_data for sample extension is incorrect
EbpfProgramType::platform_specific_data for sample extension is incorrect as it does not contain the attach type. As a result netsh ebpf load test_sample_ebpf.o is failing.

### PR DESCRIPTION
**Pull request by shankarseal**
This fixes #568 by adding sample extension attach_type to its EbpfProgramType.

Since netsh tests were missing for the sample extension, a netsh test is added in sample_ext_app.exe test app. For that some netsh test code refactoring was needed. This netsh test will run in kernel CI/CD pipeline.

### CODE REVIEW

File: tests/end_to_end/capture_helper.hpp [hunk 1]
```diff
@@ -4,6 +4,17 @@
...
+#include <fstream>
+#include <io.h>
+#include <iostream>
+#include <sal.h>
+#include <sstream>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <windows.h>
+// #include "capture_helper.hpp"
```
Comment (dthaler) [inline tests/end_to_end/capture_helper.hpp:17]: remove commented out code

File: tests/end_to_end/capture_helper.hpp [hunk 2]
```diff
@@ -28,4 +39,100 @@ class capture_helper_t final
...
+
+#define TEMPORARY_STDOUT_FILENAME ".\\stdout.txt"
+#define TEMPORARY_STDERR_FILENAME ".\\stderr.txt"
+
+std::string
+capture_helper_t::get_stdout_contents(void)
+{
+    end_capture();
+    return _stdout_contents;
+}
```
Comment (dthaler) [inline tests/end_to_end/capture_helper.hpp:54]: Putting function definitions in a .hpp causes linker problems when the header file is included by multiple c/cpp files.  These should be left in capture_helper.cpp

File: tests/end_to_end/netsh_test.cpp [hunk 3]
```diff
@@ -1,146 +1,14 @@
...
-#include "catch_wrapper.hpp"
-#include "elf.h"
 #pragma warning(push)
 #pragma warning(disable : 4200)
 #include "libbpf.h"
 #pragma warning(pop)
-#include "links.h"
-#include "maps.h"
-#include "pins.h"
+#include "netsh_test_helper.h" // Has to be above links.h
```
Comment (dthaler) [inline tests/end_to_end/netsh_test.cpp:8]: The comment implies we have a bug.  Every header should internally include any prerequisites.

File: tests/end_to_end/netsh_test_helper.h [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    _In_ DWORD dwArgCount,
+    _Inout_updates_opt_(dwTagCount) TAG_TYPE* pttTags,
+    _In_ DWORD dwTagCount,
+    _In_ DWORD dwMinArgs,
+    _In_ DWORD dwMaxArgs,
+    _Out_writes_opt_(dwArgCount - dwCurrentIndex) DWORD* pdwTagType)
+{
+    UNREFERENCED_PARAMETER(hModule);
+
+    DWORD argc = dwArgCount - dwCurrentIndex;
```
Comment (dthaler) [inline tests/end_to_end/netsh_test_helper.h:33]: .h files should not include function definitions, it causes linker problems.  Revert.

File: tests/unit/test.vcxproj [hunk 5]
```diff
@@ -204,7 +204,6 @@
     <ClCompile Include="..\..\libs\execution_context\unit\execution_context_unit_test.cpp" />
     <ClCompile Include="..\..\libs\platform\unit\platform_unit_test.cpp" />
     <ClCompile Include="..\..\libs\thunk\mock\mock.cpp" />
-    <ClCompile Include="..\end_to_end\capture_helper.cpp" />
```
Comment (dthaler) [inline tests/unit/test.vcxproj:207]: revert

File: tests/unit/test.vcxproj.filters [hunk 6]
```diff
@@ -31,9 +31,6 @@
     <ClCompile Include="..\..\libs\platform\unit\platform_unit_test.cpp">
       <Filter>Source Files</Filter>
     </ClCompile>
-    <ClCompile Include="..\end_to_end\capture_helper.cpp">
```
Comment (dthaler) [inline tests/unit/test.vcxproj.filters:34]: revert

**Comment by shankarseal** [comment]
@dthaler -- did the refactoring again. This time all function bodies are in .cpp file. Capture helper, and netsh helper (mock routines) now part of test_util.lib.

**Review by dthaler — APPROVED** [review]

**Review by Alan-Jowett — APPROVED** [review]

**Pull request merged**